reconcile/deadmanssnitch.py

```diff
@@ -0,0 +1,187 @@
+import logging
+import sys
+from typing import (
+    Any,
+    Optional,
+    cast,
+)
+
+from reconcile.gql_definitions.common.app_interface_dms_settings import (
+    DeadMansSnitchSettingsV1,
+)
+from reconcile.gql_definitions.common.clusters_with_dms import ClusterV1
+from reconcile.status import ExitCodes
+from reconcile.typed_queries.app_interface_deadmanssnitch_settings import (
+    get_deadmanssnitch_settings,
+)
+from reconcile.typed_queries.clusters_with_dms import get_clusters_with_dms
+from reconcile.utils.deadmanssnitch_api import DeadMansSnitchApi
+from reconcile.utils.runtime.integration import (
+    NoParams,
+    QontractReconcileIntegration,
+)
+from reconcile.utils.secret_reader import (
+    SecretNotFound,
+)
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.vault import (
+    VaultClient,
+    _VaultClient,
+)
+
+QONTRACT_INTEGRATION = "deadmanssnitch"
+SECRET_NOT_FOUND = "SECRET_NOT_FOUND"
+
+
+class DiffHandler:
+    UPDATE_VAULT = "update_vault"
+    CREATE_SNITCH = "create_snitch"
+    DELETE_SNITCH = "delete_snitch"
+
+    def __init__(self, deadmanssnitch_api: DeadMansSnitchApi, settings: DeadMansSnitchSettingsV1, vault_client: VaultClient = None) -> None:
+        self.deadmanssnitch_api = deadmanssnitch_api
+        self.settings = settings
+        if vault_client is not None:
+            self.vault_client = vault_client
+        else:
+            self.vault_client = cast(_VaultClient, VaultClient())
+
+    def create_diff_data(self, cluster_name: str, action: str, current_state: Optional[dict[str, Any]] = None) -> dict[str, str]:
+        data: dict[str, str] = {}
+        data["action"] = action
+        data["cluster_name"] = cluster_name
+        if action == self.UPDATE_VAULT:
+            data["snitch_url"] = current_state["check_in_url"]
+        elif action == self.DELETE_SNITCH:
+            data["token"] = current_state["token"]
+        return data
```

Comment:
@hemslo dynamic dict was chosen because each operation had its own different fields....for example to create snitch the diff object like u can see here....https://github.com/mehfuz/qontract-reconcile/blob/upstream/reconcile/test/test_deadmanssnitch.py#L172

Reply:
each action can have their own type, a base type plus 3 subtypes.